src/libs/SubscriptionUtils.ts

```diff
@@ -30,25 +119,231 @@ Onyx.connect({
+}
+
+type SubscriptionStatus = {
+    status: string;
+    isError?: boolean;
+};
+
+/**
+ * @returns The subscription status.
+ */
+function getSubscriptionStatus(): SubscriptionStatus | undefined {
+    if (hasOverdueGracePeriod()) {
+        if (hasAmountOwed()) {
+            // 1. Policy owner with amount owed, within grace period
+            if (!hasGracePeriodOverdue()) {
+                return {
+                    status: PAYMENT_STATUS.POLICY_OWNER_WITH_AMOUNT_OWED,
+                    isError: true,
+                };
+            }
+
+            // 2. Policy owner with amount owed, overdue (past grace period)
+            if (hasGracePeriodOverdue()) {
+                return {
+                    status: PAYMENT_STATUS.POLICY_OWNER_WITH_AMOUNT_OWED_OVERDUE,
+                };
+            }
+        } else {
+            // 3. Owner of policy under invoicing, within grace period
+            if (!hasGracePeriodOverdue()) {
+                return {
+                    status: PAYMENT_STATUS.OWNER_OF_POLICY_UNDER_INVOICING,
+                };
+            }
+
+            // 4. Owner of policy under invoicing, overdue (past grace period)
+            if (hasGracePeriodOverdue()) {
+                return {
+                    status: PAYMENT_STATUS.OWNER_OF_POLICY_UNDER_INVOICING_OVERDUE,
+                };
+            }
+        }
+    }
+    // 5. Billing disputed by cardholder
+    if (hasBillingDisputePending()) {
+        return {
+            status: PAYMENT_STATUS.BILLING_DISPUTE_PENDING,
+        };
+    }
+
+    // 6. Card not authenticated
+    if (hasCardAuthenticatedError()) {
+        return {
+            status: PAYMENT_STATUS.CARD_AUTHENTICATION_REQUIRED,
+        };
+    }
+
+    // 7. Insufficient funds
+    if (hasInsufficientFundsError()) {
+        return {
+            status: PAYMENT_STATUS.INSUFFICIENT_FUNDS,
+        };
+    }
+
+    // 8. Card expired
+    if (hasCardExpiredError()) {
+        return {
+            status: PAYMENT_STATUS.CARD_EXPIRED,
+        };
+    }
+
+    // 9. Card due to expire soon
+    if (hasCardExpiringSoon()) {
+        return {
+            status: PAYMENT_STATUS.CARD_EXPIRE_SOON,
+        };
+    }
+
+    // 10. Retry billing success
+    if (isRetryBillingSuccessful()) {
+        return {
+            status: PAYMENT_STATUS.RETRY_BILLING_SUCCESS,
+            isError: false,
+        };
+    }
+
+    // 11. Retry billing error
+    if (hasRetryBillingError()) {
+        return {
+            status: PAYMENT_STATUS.RETRY_BILLING_ERROR,
+            isError: true,
+        };
+    }
+
+    return undefined;
+}
+
+/**
+ * @returns Whether there is a subscription red dot error.
+ */
+function hasSubscriptionRedDotError(): boolean {
+    return getSubscriptionStatus()?.isError ?? false;
+}
+
+/**
+ * @returns Whether there is a subscription green dot info.
+ */
+function hasSubscriptionGreenDotInfo(): boolean {
+    return !getSubscriptionStatus()?.isError ?? false;
```

Comment:
This condition cause an issue [here](https://github.com/Expensify/App/issues/44804).